skbio/stats/distance/_permdisp.py

```diff
@@ -0,0 +1,269 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2013--, scikit-bio development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from functools import partial
+
+import numpy as np
+from scipy.stats import f_oneway
+from scipy.spatial.distance import cdist
+
+import hdmedians as hd
+
+from ._base import (_preprocess_input, _run_monte_carlo_stats, _build_results)
+
+from skbio.stats.ordination import pcoa
+from skbio.util._decorator import experimental
+
+
+@experimental(as_of="0.5.1")
+def permdisp(distance_matrix, grouping, column=None, test='median',
+             permutations=999):
+    """Test for Homogeneity of Multivariate Groups Disperisons using Marti
+    Anderson's PERMDISP2 procedure.
+
+    PERMDISP is a multivariate analogue of Levene's test for homogeneity of
+    multivariate variances. Non-euclidean distances are handled by reducing the
+    original distances to principle coordinates. PERMDISP calculates an
+    F-statistic to assess whether the dispersions between groups is significant
+
+
+    Parameters
+    ----------
+    distance_matrix : DistanceMatrix
+        Distance matrix containing distances between objects (e.g., distances
+        between samples of microbial communities).
+    grouping : 1-D array_like or pandas.DataFrame
```

Comment:
Are there any restrictions on the number of samples that must be in each group? Are groups with one sample in them valid, or should an error be raised?

Reply:
I don't believe we considered this, it is not reflected in the program or the tests. I can write those considerations in. I am not sure how the program will work if theres only one sample.

Reply:
I checked and groups with one sample in them are indeed valid, but the F-statistic will be nan if all groups contain only one sample